main.py

```diff
@@ -183,167 +190,122 @@ def compose_message(issues):
+        if noun is None or noun not in ["all", "ALL"]:
+            author = ctx.message.author
+            if isinstance(author, discord.Member):
+                author = author.nick or author.name
+            elif isinstance(author, discord.User):
+                author = author.name
+    else:
+        if noun in ["issue", "issues", "ISSUE", "ISSUES"]:
+            conf["pull_request"]=False
+        elif noun in ["pr", "prs", "PR", "PRs", "PRS"]:
+            conf["pull_request"]=True
+        else:
+            conf["pull_request"]=False
+            await error_channel("", "Insufficient info, defaulting to issues..")
+
+    return number_of_issues, conf, author
+
+
+for name, conf in command_config.items():
+    @bot.command(name=name)
+    async def magic_function(ctx, number_of_issues: int, noun=None, channel=None, name=name, conf=conf): # name=name and conf=conf used to prevent late binding
+        if channel is None:
+            channel=ctx.channel
+        error_channel = error_handler(channel)
+
+        number_of_issues, conf, author=await modify_params(ctx=ctx, conf=conf, name=name, number_of_issues=number_of_issues, noun=noun, channel=channel, error_channel=error_channel)
+
+        await set_playing('On The Cue')
+        async with channel.typing():
+            issues = [x async for x in get_issues(**conf, error_channel=error_channel)]
+            reply = discord.Embed(color=discord.Color.purple())
+
+            chosen_issues=[]
+            if name=="rand":
+                """
+                @TODO: improve algorithm to save queries
+                1. get first batch of github_max (100), find total number
+                2. if total < requested, return all
+                3. Generate requested random numbers
+                4. get issues till max(generated_list)
+                5. return them
+                """
+                chosen_issues = random.choices(issues, k=number_of_issues)
+                reply.title = f"{number_of_issues} random picks out of {len(issues)}:"
+
+            elif name in ["rq", "fq"]:
+                # the author check is only for review, author=None for feedback
+                if author:
+                    issues = pr_with_pending_review(get_pr_details(issues), author)
+                    # since async islice doesn't exist
+                    async for issue in issues:
+                        chosen_issues.append(issue)
+                        if len(chosen_issues) == number_of_issues:
+                            break
+                else:
+                    chosen_issues=issues[:number_of_issues]
+                kind = "PR(s)" if conf["pull_request"] else "issue(s)"
+                queue_type = "review" if name=="rq" else "feedback"
+                selection = f"for @{author}" if author else f"in {queue_type} queue"
+                reply.title = f"Oldest {number_of_issues} {kind} {selection}:"
+            
```

Comment:
Some minor refactoring so we can do the following:
```suggestion
            chosen_issues, reply = command_function[name](author, issues, reply)
```

Reason: deep indentations, long function length

Reply:
See bd2e093
It'd be better for the choosing parts to be in different functions, rather if-else, as pointed out by you earlier.